### PR TITLE
adds node config status primitives for common use

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/1918.

### CODE REVIEW

File: pkg/apis/core/v1alpha1/node_funcs.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package v1alpha1
+
+const (
+	NodeConfigStatusStatusTrue  = "True"
+	NodeConfigStatusTypeDrained = "Drained"
+)
+
+func (s NodeConfigStatus) HasFinalConditiion() bool {
```
Comment (xh3b4sd) [inline pkg/apis/core/v1alpha1/node_funcs.go]: We already have this in the [aws-operator](https://github.com/giantswarm/aws-operator/blob/4ed80749c47f24153e539f3c6f0d088a0f535a73/service/awsconfig/v10/resource/lifecycle/create.go#L302-L310). We will need this for KVM and Azure as well, so moving to here for common use.

File: pkg/apis/core/v1alpha1/node_funcs.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	for _, c := range s.Conditions {
+		if c.Type == NodeConfigStatusTypeDrained && c.Status == NodeConfigStatusStatusTrue {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (s NodeConfigStatus) NewFinalConditiion() NodeConfigStatusCondition {
```
Comment (xh3b4sd) [inline pkg/apis/core/v1alpha1/node_funcs.go]: We have this in the [node-operator](https://github.com/giantswarm/node-operator/blob/5530790c9ada231704ee296411e2d988585bc41b/service/nodeconfig/v1/resource/node/create.go#L163-L166). When we move the condition status checks to here it makes sense to have the magic strings in one place. So moving to here for the common ground.

**Review by MarcelMue — APPROVED** [review]
LGTM

File: pkg/apis/core/v1alpha1/node_funcs.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	return NodeConfigStatusCondition{
+		Status: NodeConfigStatusStatusTrue,
+		Type:   NodeConfigStatusTypeDrained,
+	}
+}
```
Comment (kopiczko) [inline pkg/apis/core/v1alpha1/node_funcs.go]: I don't think we should put any custom logic in this repo. We have repos like certs and radomkeys where we put the logic. This here isn't a big deal but let's be careful with adding things like that to apiextensions.

Comment (xh3b4sd) [reply pkg/apis/core/v1alpha1/node_funcs.go]: I hear you but I do not see how this is super specific. This is super generic. Everybody can use it to update the node config status to the final drained status and to detect such states.

Comment (kopiczko) [reply pkg/apis/core/v1alpha1/node_funcs.go]: By custom I mean not manipulating CRs in the K8s api. Let's leave it like it is, but if it grows I think we should extract the repo with shared logic.

Comment (xh3b4sd) [reply pkg/apis/core/v1alpha1/node_funcs.go]: Do you mean that there should only be type definitions and generated API clients in here?

Comment (kopiczko) [reply pkg/apis/core/v1alpha1/node_funcs.go]: That's my feeling. I think that's how kubernetes packages are designed.

Comment (xh3b4sd) [reply pkg/apis/core/v1alpha1/node_funcs.go]: I see. The line is a bit blurry here but I get you. Lets see how this goes.

**Pull request merged**